src/main/java/com/raisetech10/liquor_management/service/LiquorServiceImpl.java

```diff
@@ -19,14 +21,51 @@ public LiquorServiceImpl(LiquorMapper liquorMapper) {
 
     //GET
     @Override
-    public List<Liquor> findAll(){
+    public List<Liquor> findAll() {
         return liquorMapper.findAll();
     }
+
+    //GET Exception
+    @Override
+    public Liquor findById(int id) {
+        Optional<Liquor> liquor = this.liquorMapper.findById(id);
+        return liquor.orElseThrow(() -> new ResourceNotFoundException("★Liquor not found★" + id));
+    }
+
     //POST
     @Override
     public Liquor createLiquor(Liquor liquor) {
         liquorMapper.insert(liquor);
         return liquor;
     }
 
+    //PATCH
+    @Override
+    public Liquor updateLiquor(int id, String liquorType, String producingCountry, String liquorName, Integer alcoholContent) {
+        Liquor liquor = liquorMapper.findLiquorId(id)
+                .orElseThrow(() -> new ResourceNotFoundException("★Liquor not found★" + id));
+        if (liquorType != null) {
+            liquor.setLiquorType(liquor.getLiquorType());
```

Comment:
これだと、DBから取得したliquorからliquorTypeを取得して、同じliquorにセットし直す処理になってしまい、値が更新されないです！引数のliquorTypeをセットするようにしてください！
```suggestion
            liquor.setLiquorType(liquorType);
```


気付けるポイントとしては、引数の変数が使われていないので薄くなっていないでしょうか？？
定義しているのに使われていない変数は薄くなるので、どこかで使ってあげる or 削除してあげた方が良いですね！

Reply:
上記、変更したら更新できるようになりました！